### DISCUSSION TITLE
conflict with permissions and custom policies

### DISCUSSION THREAD

**Discussion by cesarZaabra** [spatie/laravel-permission#2372] [Q&A]
I am currently using Laravel-permission for access management in my Laravel project, but I have a problem, I need to generate a "SuperAdmin" user that consists of passing all access permissions but I also have registered policies that validate other factors

My problem is that I need the user to pass all the permissions of Laravel-permission but to respect the custom policies that were generated.

**Comment by parallels999** [comment]
Hi, you do not offer the necessary information to review
https://github.com/spatie/laravel-permission/blob/c7e0eb2851bcc0bdb42fd82d25ff8236aec463b8/.github/ISSUE_TEMPLATE/bug_report.md?plain=1#L11-L42

**Comment by cesarZaabra** [comment]
Hi.

My permissions with laravel-permission are
* post create
* post edit
* post-show

and I have a policy created

```php
namespace App\Policies;
use App\Models\Post;
use App\Models\User;
use Illuminate\Auth\Access\HandlesAuthorization;
class PostPolicy
{
    use HandlesAuthorization;
    /**
     * @param User $user
     * @param Post $post
     * @return bool
     */
    public function edit(User $user, Post $post): bool
    {
        return $post->user_id == $user->id;
    }
}
```

When you create a super admin user, it is necessary to be able to perform all the permissions regardless of whether you have access or not, it was solved with a gate in "app/Providers/AuthServiceProvider.php"

```php
Gate::before(function (User $user, $ability) {
            return $user->isSuperAdmin() ? true:null;
        });
```

The problem is that the superAdmin can perform all the functions with or without access, but it also gives him the possibility to edit the post of other users and this should not be allowed.

**Comment by drbyte** [comment]
It's possible you require also a `Gate::after()` rule.

See: https://spatie.be/docs/laravel-permission/v5/basic-usage/super-admin#content-gateafter
and the Laravel docs for `Gate::after()`
